web/src/lib/components/user-settings-page/PincodeInput.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="ts">
+  interface Props {
+    label: string;
+    value?: string;
+    pinLength?: number;
+    tabindexStart?: number;
+  }
+
+  let { label, value = $bindable(''), pinLength = 6, tabindexStart = 0 }: Props = $props();
+
+  let pinValues = $state(Array.from({ length: pinLength }).fill(''));
+  let pincodeInputElements: HTMLInputElement[] = $state([]);
+
+  export function reset() {
```

Comment:
Components shouldn't export any functions. If you need this use events or bind to a prop.